File: PingSysInventory/PingSysInventory.Web/Modules/Inventory/Forms/CategoriesColumns.cs
using PingSysInventory.Inventory;

namespace PingSysInventory.Inventory.Forms;

[ColumnsScript("Inventory.Categories")]
[BasedOnRow(typeof(CategoriesRow), CheckNames = true)]
public class CategoriesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CategoryId { get; set; }

    [EditLink, Width(250)]
    public string CategoryName { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}

File: PingSysInventory/PingSysInventory.Web/Modules/ServerTypes/Inventory/CategoriesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoriesRow } from "./CategoriesRow";

export interface CategoriesColumns {
    CategoryId: Column<CategoriesRow>;
    CategoryName: Column<CategoriesRow>;
    IsActive: Column<CategoriesRow>;
}

export class CategoriesColumns extends ColumnsBase<CategoriesRow> {
    static readonly columnsKey = 'Inventory.Categories';
    static readonly Fields = fieldsProxy<CategoriesColumns>();
}